packages/rum-core/src/domain/waitPageActivityEnd.spec.ts

```diff
@@ -41,38 +42,56 @@ function eventsCollector<T>() {
   }
 }
 
+const RUM_CONFIGURATION: RumConfiguration = {
+  ...validateAndBuildRumConfiguration({
+    clientToken: 'xxx',
+    applicationId: 'AppId',
+    trackResources: true,
+    trackLongTasks: true,
+  })!,
+  ...SPEC_ENDPOINTS,
+}
+
 describe('createPageActivityObservable', () => {
   const { events, pushEvent } = eventsCollector<PageActivityEvent>()
 
-  let setupBuilder: TestSetupBuilder
+  const lifeCycle = new LifeCycle()
+  const domMutationObservable = new Observable<void>()
   let pageActivitySubscription: Subscription
   let notifyPerformanceEntries: (entries: RumPerformanceEntry[]) => void
 
+  function startListeningToPageActivities(
+    extraConfiguration: Partial<RumConfiguration> = { excludedActivityUrls: [EXCLUDED_FAKE_URL] }
+  ) {
+    pageActivitySubscription?.unsubscribe()
```

Comment:
**💬 suggestion:** with your approach, we can have multiple `startListeningToPageActivities` calls during a single test case, that's why we need to `unsubscribe` here.

This is a bit unexpected and makes things more complicated than needed. What about removing the call to `startListeningToPageActivities` from `beforeEach` and call it once in each test case instead, like we did with `setupBuilder.build()`?